loadtest/control/gencontroller/actions.go

```diff
@@ -204,19 +208,60 @@ func (c *GenController) createPost(u user.User) control.UserActionResponse {
 		}
 	}
 
+	if hasFilesAttached {
+		if err := control.AttachFilesToPost(u, post); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+	}
+
 	postId, err := u.CreatePost(post)
 	if err != nil {
-		st.dec("posts")
+		st.dec(StateTargetPosts)
 		return control.UserActionResponse{Err: control.NewUserError(err)}
 	}
+
 	if shouldLongThread {
 		st.setLongRunningThread(postId, channel.Id, channel.TeamId)
 	}
 	return control.UserActionResponse{Info: fmt.Sprintf("post created, id %v", postId)}
 }
 
+func (c *GenController) createPostReminder(u user.User) control.UserActionResponse {
+	if !st.inc(StateTargetPostReminders, c.config.NumPostReminders) {
+		return control.UserActionResponse{Info: "target number of post reminders reached"}
+	}
+
+	team, err := u.Store().RandomTeam(store.SelectMemberOf)
+	if err != nil {
+		st.dec(StateTargetPostReminders)
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	ch, err := u.Store().RandomChannel(team.Id, store.SelectMemberOf)
+	if err != nil {
+		st.dec(StateTargetPostReminders)
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	post, err := u.Store().RandomPostForChannel(ch.Id)
```

Comment:
I kept getting these errors while running the PR:

```
followThread loadtest/control/gencontroller/actions.go:469 memstore: post not found caller="loadtest/loadtest.go:57" controller_id=20 user_id=wd846e8uppbym8b1ugooyqcoew
createPostReminder loadtest/control/gencontroller/actions.go:249 memstore: post not found caller="loadtest/loadtest.go:57" controller_id=13 user_id=3s5jwrobk3yd3qb77q9tn5hj6c
```

I am wondering if we should just use `Store().RandomPost()` to directly get a post? It's essentially the same, but guaranteed to return a post (atleast after the first post is created).

Lastly, we should also check for `ErrPostNotFound` separately, and return Info, not Err if that's the case. See: https://github.com/mattermost/mattermost-load-test-ng/pull/593/files#diff-748b06ff2082afa449f1e1e3a37ce5f6986be5511b52d9e5545e1ba5504e6266R262

Reply:
Looking into it a bit more, followThread has `UpdateThreadFollow` which needs the teamID. To do that, we just need to fetch the channelID from the post, and thereby the teamID. One catch there is there is no teamID for DM/GM, and then `u.UpdateThreadFollow` throws an error. I am not sure how the actual client does this. Something to look into.

Reply:
I realized that we had the RandomPost function after I had written the RandomTeam/RandomChannel/RandomPostForChannel logic, but since we needed the team ID, I decided to leave it as it was.

> One catch there is there is no teamID for DM/GM, and then u.UpdateThreadFollow throws an error. I am not sure how the actual client does this. Something to look into.

Oh, good catch, I had not taken this into account. I'll review the whole thing, thanks!

Reply:
Ended up modifying RandomPost to receive a SelectionType, so we can filter the type of channel the post lives in. Please check https://github.com/mattermost/mattermost-load-test-ng/pull/594/commits/6eaffc7e89ea1aea313a1c349503392161673cba and https://github.com/mattermost/mattermost-load-test-ng/pull/594/commits/a240e85695b59cecb496a3d77aca5d897a27ff2a :)